orchestra/models/communication/mixins.py

```diff
@@ -66,3 +67,35 @@ def __str__(self):
             self.is_available,
             self.is_winner
         )
+
+    def _mark_staffbot_request_complete(self, request):
+        from orchestra.models import StaffBotRequest
+        request.status = StaffBotRequest.Status.COMPLETE.value
+        request.save()
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        from orchestra.models import StaffingRequestInquiry
+        request = self.request_inquiry.request
+        if self.is_winner:
+            self._mark_staffbot_request_complete(request)
+        else:
+            inquiries = (
+                StaffingRequestInquiry.objects.filter(request=request)
+            ).distinct()
+            num_inquired_workers = len(
+                set(inquiries.values_list(
+                    'communication_preference__worker__id', flat=True)
+                    )
+            )
+
+            responded_inquiries = inquiries.filter(
+                responses__isnull=False).distinct()
+            num_responded_workers = len(
+                set(responded_inquiries.values_list(
+                    'communication_preference__worker__id', flat=True)
+                    )
+            )
+
+            if num_responded_workers >= num_inquired_workers:
+                self._mark_staffbot_request_complete(request)
```

Comment:
this is nice cleanup. do we need a data migration to do this for all old projects, or will some other logic mark them as complete eventually?